### PR TITLE
Add original code

### PR DESCRIPTION
**Pull request by lucasxiecelestica**
Add original main.py  code

### CODE REVIEW

**Review by sandycelestica — APPROVED** [review]

**Review by sandycelestica — COMMENTED** [review]
pass

File: main.py [hunk 1]
```diff
@@ -1779,12 +1786,13 @@ def ssd_firmwareinfo(device):
 
     checkin = 0
     testname="SSD Firmwareinfo Test"
+	# get the SSD information by call the smartctl cmd
```
Comment (grantshao) [inline main.py]: align with  upstairs code

File: main.py [hunk 2]
```diff
@@ -2350,7 +2369,6 @@ def check_pcie_speed(device = {}):
                         checkin = 0
```
Comment (grantshao) [inline main.py:2369]: align with upstair code.

**Review by grantshao — APPROVED** [review]

File: main.py [hunk 3]
```diff
@@ -2278,7 +2295,7 @@ def ssd_all(device):
     #"""Read SSD temperature range 0~70 �C"""
     checkin = 0
```
Comment (grantshao) [inline main.py:2296]: leave a blank space before and after '='

File: main.py [hunk 4]
```diff
@@ -2254,7 +2271,7 @@ def ssd_all(device):
     #"""Check the later bad block status which may created"""
     checkin = 0
     testname="SSD Badblock Test"
```
Comment (grantshao) [inline main.py]: leave a blank space before and after '='

File: main.py [hunk 5]
```diff
@@ -2063,14 +2077,15 @@ def ssd_badblock(device):
...
 #
 
 @cli.command()
 @click.argument("device")
 def ssd_temperature(device):
-    """Read SSD temperature range 0~70 �C"""
+    """Read SSD temperature �C"""
 
     checkin = 0
     testname="SSD Temperature Test"
```
Comment (grantshao) [inline main.py]: leave a blank space before and after '='

File: main.py [hunk 6]
```diff
@@ -2132,7 +2148,7 @@ def ssd_all(device):
     #"""Show ssd fwinfo"""
     checkin = 0
     testname="SSD Firmwareinfo Test"
```
Comment (grantshao) [inline main.py]: leave a blank space before and after '='

File: main.py [hunk 7]
```diff
@@ -2147,7 +2163,7 @@ def ssd_all(device):
     checkin = 0
     testname="SSD Capcity Test"
```
Comment (grantshao) [inline main.py]: leave a blank space before and after '='

File: main.py [hunk 8]
```diff
@@ -2161,7 +2177,7 @@ def ssd_all(device):
     #"""Show ssd serial number"""
     checkin = 0
     testname="SSD SN Test"
```
Comment (grantshao) [inline main.py]: leave a blank space before and after '='

File: main.py [hunk 9]
```diff
@@ -2175,7 +2191,7 @@ def ssd_all(device):
     #"""Show ssd Remaining Time"""
     checkin = 0
     testname="SSD RemainTime Test"
```
Comment (grantshao) [inline main.py]: leave a blank space before and after '='

File: main.py (hunk 4 above)
Comment (grantshao) [inline main.py]: leave a blank space before and after '='

**Comment by grantshao** [comment]
pass

**Comment by mudsut4ke** [comment]
Pass,
But I think the format of output should be a format like

show ssd all /dev/sda1
not
show ssd_all /dev/sda1

But you can submit to Banyan to let them review first , maybe they will accept this format

**Pull request merged**